packages/synapse-react-client/src/components/DataGrid/DataGridUtils.ts

```diff
@@ -27,3 +27,30 @@ export function rowsAreIdentical(a: DataGridRow, b: DataGridRow): boolean {
   }
   return true
 }
+
+export function removeNoOpOperations(
```

Comment:
Factored out of SynapseGrid and updated to filter out no-ops and return the valid operations.